gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/util/EhrResourceExtractorUtil.java

```diff
@@ -65,6 +65,34 @@ public static boolean hasEhrFolder(RCMRMT030101UK04Component component) {
         return component.getEhrFolder() != null;
     }
 
+    public static RCMRMT030101UK04EhrComposition extractEhrCompositionForNarrativeStatement(RCMRMT030101UK04EhrExtract ehrExtract,
```

Comment:
This might possibly throw an error if the narrative statement is coming from a nested compound statement, as it wont be able to find the matching narrative statement id

Reply:
if he uses my suggestion for rewriting his mapping method this should no longer be an issue.